File: frontend/src/types/session.ts
/** Session-related TypeScript interfaces matching the API contract. */

export interface CreateSessionRequest {
  participant_name?: string;
  metadata?: Record<string, string>;
}

export interface QuestionPayload {
  question_id: string;
  text: string;
  question_number: number;
}

export interface SessionResponse {
  session_id: string;
  user_id: string;
  survey_id: string;
  status: "active" | "completed" | "exited";
  current_question: QuestionPayload;
  question_number: number;
  max_questions: number;
  created_at: string;
}

export interface SubmitAnswerRequest {
  answer: string;
  question_id?: string;
  question_text?: string;
}

export interface NextQuestionResponse {
  session_id: string;
  status: "active" | "completed";
  question?: QuestionPayload;
  completion_reason?: string;
  question_number: number;
  max_questions: number;
}

export interface ConversationEntry {
  question_id: string;
  question_text: string;
  answer_text: string;
  question_number: number;
  answered_at: string;
}

export interface SessionDetailResponse {
  session_id: string;
  user_id: string;
  survey_id: string;
  status: "active" | "completed" | "exited";
  conversation: ConversationEntry[];
  question_count: number;
  created_at: string;
  completed_at?: string;
}

export interface SessionCompleteResponse {
  session_id: string;
  status: "exited";
  question_count: number;
  message: string;
}

export interface ResponseListResponse {
  responses: SessionDetailResponse[];
  total: number;
  skip: number;
  limit: number;
}

File: backend/app/schemas/session.py
"""Session Pydantic schemas."""

from typing import Dict, List, Optional

from pydantic import BaseModel, Field


class CreateSessionRequest(BaseModel):
    """Request schema for creating a session."""

    participant_name: Optional[str] = None
    metadata: Dict[str, str] = Field(default_factory=dict)


class QuestionPayload(BaseModel):
    """Question data payload."""

    question_id: str
    text: str
    question_number: int


class SessionResponse(BaseModel):
    """Response schema after creating a session."""

    session_id: str
    user_id: str
    survey_id: str
    status: str
    current_question: QuestionPayload
    question_number: int
    max_questions: int
    created_at: str


class NextQuestionResponse(BaseModel):
    """Response schema for the next question or completion."""

    session_id: str
    status: str
    question: Optional[QuestionPayload] = None
    completion_reason: Optional[str] = None
    question_number: int
    max_questions: int


class ConversationEntry(BaseModel):
    """A single Q&A entry in the conversation."""

    question_id: str
    question_text: str
    answer_text: str
    question_number: int
    answered_at: str


class SessionDetailResponse(BaseModel):
    """Detailed session response with conversation history."""

    session_id: str
    user_id: str
    survey_id: str
    status: str
    conversation: List[ConversationEntry]
    question_count: int
    created_at: str
    completed_at: Optional[str] = None


class SessionCompleteResponse(BaseModel):
    """Response schema for exiting a session."""

    session_id: str
    status: str
    question_count: int
    message: str
